libshaderc/include/shaderc.hpp

```diff
@@ -125,6 +126,35 @@ class CompileOptions {
     shaderc_compile_options_set_generate_debug_info(options_);
   }
 
+  // Define an includer interface, on which libshaderc will invoke the methods
+  // to resolve the full path and content of the file to be included, and clean
+  // up the resources used for the including process. Client code should
+  // implement this interface and configure the includer instance then set it to
+  // libshaderc through following api.
+  class Includer {
+   public:
+    virtual shaderc_includer_response* GetIncluderResponse(
```

Comment:
If you want to keep the current solution, please rename Includer -> IncluderInterface
That's more in line with the advice in the Google style guide for permitted uses of multiple inheritance (it may be that a client uses multiple inheritance as part of this solution)